assets/src/edit-story/app/story/useStoryReducer/test/restore.js

```diff
@@ -149,4 +149,47 @@ describe('restore', () => {
       story: {},
     });
   });
+
+  it('should restore with existing current page (if it exists in restore story) if no current page supplied', () => {
+    // Initialize state with a current story page.
+    const currentPage = { current: '222' };
+    const reducer = setupReducer(currentPage);
+    const { restore } = reducer;
+
+    // Restore state to hold story
+    const storyWithNoCurrentPage = {
+      pages: [{ id: '111' }, { id: '222', elements: [{ id: '333' }] }],
+      selection: ['333'],
+      story: { a: 1 },
+    };
+    const result = restore(storyWithNoCurrentPage);
+
+    expect(result).toStrictEqual({
+      animationState: STORY_ANIMATION_STATE.RESET,
+      ...storyWithNoCurrentPage,
+      ...currentPage,
+    });
+  });
+
+  it('should restore with first page as current if initialized with faulty current pageId', () => {
+    // Initialize state with a story page that does not exist
+    const currentPage = { current: 'DOES NOT EXIST' };
+    const reducer = setupReducer(currentPage);
+    const { restore } = reducer;
+
+    // Restore state to hold story
+    const storyWithNoCurrentPage = {
+      pages: [{ id: '111' }, { id: '222', elements: [{ id: '333' }] }],
+      selection: ['333'],
+      story: { a: 1 },
+    };
+    const result = restore(storyWithNoCurrentPage);
+
+    // Should set first story as current
+    expect(result).toStrictEqual({
+      animationState: STORY_ANIMATION_STATE.RESET,
+      ...storyWithNoCurrentPage,
+      current: '111',
+    });
+  });
```

Comment:
@barklund just added. is this what you were thinking or something else?

Reply:
@littlemilkstudio , that's perfect.

And I just wanted to make sure, that if you e.g. copy-pasted the URL with a deep-link to a page, but left out a character, wouldn't result in a broken editor. Defaulting to the first page seems like a fair choice.

This test confirms that (on a lower level, but good enough).